### PR TITLE
Added internal implementations

### PR DESCRIPTION
**Pull request by f7deleon**
Added internal implementations so testing becomes easier.

### CODE REVIEW

File: core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            )
+        )
+      } catch (io: IOException) {
+        continuation.resumeWithException(io)
+      } catch (illegalArgument: IllegalArgumentException) {
+        continuation.resumeWithException(illegalArgument)
+      }
+    }
+  }
+}
```
Comment (agussante16) [inline core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt:220]: nl

**Review by agussante16 — APPROVED** [review]

File: core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+import java.io.IOException
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+internal class SuspendedLocationImpl(
+    internal var context: Context,
+    internal var locationServices: FusedLocationProviderClient) {
```
Comment (matir91) [inline core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt:31]: I think `) {` in a new line is better

File: core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+  )
+  internal suspend fun requestCurrentLocation(
+      runnerContext: CoroutineContext = Dispatchers.Default
+  ): Location = withContext(runnerContext) {
+    suspendCoroutine { continuation: Continuation<Location> ->
+      locationServices
+          .requestLocationUpdates(
+              LocationRequest().setInterval(1),
+              object : LocationCallback() {
+                override fun onLocationResult(locationResult: LocationResult) {
```
Comment (matir91) [inline core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt:95]: maybe in another pr, but I'd refactor it. I'd use flow here

**Review by matir91 — APPROVED** [review]

File: core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt (hunk 3 above)
Comment (f7deleon) [reply core/src/main/java/com/xmartlabs/suspendedlocation/core/SuspendedLocationImpl.kt:95]: we allowed both one with coroutines and other with flow

**Pull request merged**